components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs

```diff
@@ -169,7 +169,24 @@ IJsonSerializer jsonSerializer
                     foreach (var assignedJob in _assignedJobs) {
 
                         if (diagInfo.Id == assignedJob.Value.Job.Id) {
-                            _publisherDiagnosticInfo.AddOrUpdate(assignedJob.Value.Job.Id, diagInfo);
+                            foreach (var entry in _publishedNodesEntries) {
+                                var id = _publishedNodesJobConverter.
```

Comment:
Now lookin at this again I understand that `jobId` is going to be determined only by the following fields:

* `EndpointUrl`
* `UseSecurity`
* `OpcAuthenticationMode` together with `OpcAuthenticationUsername` and `OpcAuthenticationPassword` 
* `DataSetWriterGroup`

So `DataSetWriterId` and `DataSetPublishingInterval` do not influence `jobId`. This makes sense given that only the above fields are going to define `jobId` and corresponding session details. And fields such as `DataSetWriterId` and `DataSetPublishingInterval` are only going to affect names of subscriptions within the session. What this means is that there might be several entries in `_publishedNodesEntries` list which would result in the same `jobId`. So only taking the first one does not seem to be the correct approach, neither does taking all of them. So maybe we should take only the fields whose values influence the `jobId`, meaning that we exclude `DataSetWriterId` and `DataSetPublishingInterval` from the assignment bellow. Then all endpoint definitions that match on the fields that we take would be part of the diagnostics info. @dacolgit, @cristipogacean, what you guys think about this ?

The other problem that we have here is that we are iterating over `_publishedNodesEntries` now, which means that we have to use `_lockConfig` together with `_lockJobs` instead of only having `_lockJobs` above on the line `117`. This is problematic because we introduced `_lockJobs` so that heartbeat reporting can happen faster and not be affected as much by direct method calls. @cristipogacean, is there a way to get the meaningful fields that I've mentioned above from the job definition itself ?

Reply:
Extended the messageTrigger to export the following info:
- `EndpointUrl
- UseSecurity
- OpcAuthenticationMode together with OpcAuthenticationUsername and OpcAuthenticationPassword
- DataSetWriterGroup`

As discussed the `DataSetWriterId` and 'DataSetPublishingInterval' are removed from the 'EndpointDiagnosticInfo'